### PR TITLE
Added yarn lint as presubmit action

### LINKED ISSUE #197: Add ESLint support

### PR DESCRIPTION
**Pull request by kebrahim**
Fixes #197 

- [x] Tests pass
- [x] Appropriate changes to README are included in PR
- [x] Feature has been fleshed out in design document

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -26,8 +26,8 @@ on:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  # This workflow contains a single job called "java-format"
+  java-format:
```
Comment (egracer) [inline .github/workflows/linting.yml:30]: Out of curiosity, why is this a separate workflow as opposed to being another job in the backend workflow?

File: callmeback/frontend/.eslintrc.json [hunk 2]
```diff
@@ -22,6 +22,9 @@
...
         "jsx-a11y"
     ],
-    "rules": {
+    "settings": {
+        "react": {
+            "version": "detect"
+        }
     }
-}
+}
```
Comment (egracer) [inline callmeback/frontend/.eslintrc.json:30]: Vaguely recall that we are ending files with newlines, is that right?

File: .github/workflows/linting.yml (hunk 1 above)
Comment (kebrahim) [reply .github/workflows/linting.yml:30]: i honestly don't know. it's been like that for awhile. we could probably combine them (which we've done in maslow-experimental), but it doesn't seem necessary for this repo

File: callmeback/frontend/.eslintrc.json (hunk 2 above)
Comment (kebrahim) [reply callmeback/frontend/.eslintrc.json]: done

**Pull request merged**